src/redisraft.c

```diff
@@ -679,6 +679,11 @@ static void handleRedisCommand(RedisRaftCtx *rr,
         return;
     }
 
+    /* reset asking state when we go to execute the command */
```

Comment:
You should reset the asking state *after* handling the current command.

Reply:
Also, line 669 and line 677 contradict each other?